python/rmm/_lib/memory_resource.pxd

```diff
@@ -21,11 +21,11 @@ from libcpp.vector cimport vector
 cdef extern from "rmm/mr/device/device_memory_resource.hpp" \
         namespace "rmm::mr" nogil:
     cdef cppclass device_memory_resource:
-        pass
+        void* allocate(size_t bytes) except +
+        void deallocate(void* ptr, size_t bytes) except +
```

Comment:
AFAICT these methods are being added just for the purpose of the test. I'm not opposed to exposing these functions in the Python API, but that seems like it merits discussion beyond this largely unrelated PR.

Reply:
I'm also fine if you want to push back here and just get this done, but I wanted to at least have that discussion recorded if so.

Reply:
I think it's probably a good idea to expose these anyway for users that want to use RMM but not necessarily construct memory owning DeviceBuffers. Also see my comment [below](https://github.com/rapidsai/rmm/pull/980#discussion_r850354608).

Reply:
I added a test for `allocate/deallocate`